### PR TITLE
[SW-286, SW-297, SW-323] Limit amount input to numbers and "."

### PR DESCRIPTION
**Pull request by pstachel**

### CODE REVIEW

**Review by peronczyk — CHANGES_REQUESTED** [review]

**Review by konradodo — APPROVED** [review]

File: src/popup/components/InputField.vue [hunk 1]
```diff
@@ -158,11 +158,20 @@ export default {
     },
   },
   methods: {
+    checkModifierKey(event) {
```
Comment (peronczyk) [inline src/popup/components/InputField.vue:161]: In my opinion there is no need to create separate method for this logic as it's used only once.

File: src/popup/components/InputField.vue [hunk 2]
```diff
@@ -158,11 +158,20 @@ export default {
...
+      return ['ctrlKey', 'metaKey'].some((key) => event[key]);
+    },
     checkIfNumber(event) {
-      if (this.type === 'number' && !['Enter', 'Backspace'].includes(event.key) && !/^[0-9/-]$/.test(event.key)) {
+      if (this.type !== 'number' || event.key.length > 1 || this.checkModifierKey(event)) return;
+      if ((this.value?.includes('.') && [',', '.'].includes(event.key)) || !/^([0-9]+|,|\.)$/.test(event.key)) {
```
Comment (peronczyk) [inline src/popup/components/InputField.vue:166]: This should be `else if`.

File: src/popup/components/InputAmountV2.vue [hunk 3]
```diff
@@ -4,6 +4,7 @@
     class="input-amount"
     type="number"
     placeholder="0.00"
+    inputmode="decimal"
```
Comment (peronczyk) [inline src/popup/components/InputAmountV2.vue:7]: In my opinion this should be moved into InputField component and be based on input type.

File: src/popup/components/InputAmount.vue [hunk 4]
```diff
@@ -11,6 +11,7 @@
     :message="$attrs['message'] || errors.first('amount')"
     class="input-amount"
     type="number"
+    inputmode="decimal"
```
Comment (peronczyk) [inline src/popup/components/InputAmount.vue:14]: In my opinion this should be moved into InputField component and be based on input type.

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/popup/components/InputField.vue [hunk 5]
```diff
@@ -159,10 +163,16 @@ export default {
   },
   methods: {
     checkIfNumber(event) {
-      if (this.type === 'number' && !['Enter', 'Backspace'].includes(event.key) && !/^[0-9/-]$/.test(event.key)) {
+      if (!(this.type !== 'number' || event.key.length > 1 || ['ctrlKey', 'metaKey'].some((key) => event[key]))
```
Comment (peronczyk) [inline src/popup/components/InputField.vue:166]: With the multiline if statements I suggest to do the same approach as with the object properties. So we are able to do this:
```
{ a: 1, b: 2, c: 3 }
```
and this:
```
{
  a: 1,
  b: 2,
  c: 3
}
```
Nothing in between, so this is wrong:
```
{ a: 1, b: 2,
c: 3 }
```
I think the same rule should be applied to if statements. So I suggest to do it as follows:
```
if (
  (test1 || test2 || test3)
  && (test4 || test5)
  && test6
) {
  ...
}
```

**Review by peronczyk — CHANGES_REQUESTED** [review]

**Review by peronczyk — APPROVED** [review]

**Pull request merged**